### ISSUE TITLE
KL Divergence calculation

### ISSUE THREAD

**Issue by minhkhoale** [psanch21/causal-flows#3]
Hi there; thank you for your excellent work. I have a question about the KL Divergence calculation. In the main paper, you mentioned that the KL divergence of observational distribution is calculated between p_M and p_\theta, which are the true and the learned distribution of exogenous. However, I saw in the code that KL is calculated using true and learned endogenous distributions, p_x. So, I don't know if I misunderstood your point or not. Thank you.

**Comment by psanch21** [comment]
Hi @ktncktnc,

Thank you for your interest in our work! I think there might be a misunderstanding: p_\theta and p_M are distributions over x, the endogenous variables. If this isn't clearly stated in the paper or the code, please let me know where specifically (including line numbers or sections), and we'll work to provide further clarification.

**Comment by minhkhoale** [comment]
Thank you for your reply. In Figure 3 in the paper, P_x is the distribution over x, so I think p_M and p_\theta should be distribution over u. Please correct me if I am wrong.
![image](https://github.com/psanch21/causal-flows/assets/25315522/793ac089-a765-4092-a2be-14df2191e77b)

**Comment by adrianjav** [comment]
Hi @ktncktnc , there seems to be a mismatch between notation on sections 3 and 6.

1. Indeed, $P_\theta$ and $P_\mathcal{M}$ in Figure 3 are the distributions over the exogeneous variables, $u$.
2. The part I believed confused you was this one in section 6:
![image](https://github.com/psanch21/causal-flows/assets/25713831/051179f3-daef-41b2-ad63-58b73cd5ac4b)
There, we were referring to the distributions over $x$ induced by the true SCM and the causal normalizing flow.

TL;DR: We used the same notation for distributions on $u$ and $x$ without realizing. 

I hope this helps!
